.github/actions/dependencies/action.yml

```diff
@@ -11,23 +13,33 @@ runs:
       run: |
         conan export --version 1.1.10 external/snappy
         conan export --version 4.0.3 external/soci
-    - name: add Ripple Conan remote
-      if: env.CONAN_URL != ''
+    - name: add Conan remotes
+      if: env.CONAN_REMOTE_URL != ''
       shell: bash
       run: |
-        if conan remote list | grep -q "ripple"; then
-            conan remote remove ripple
-            echo "Removed conan remote ripple"
+        echo "Adding Conan remotes for recipes."
+        if conan remote list | grep -q "recipes"; then
+          conan remote update recipes --index 0 --url ${{ env.CONAN_REMOTE_URL }}/recipes
+          echo "Updated Conan remote 'recipes' to ${{ env.CONAN_REMOTE_URL }}/recipes."
+        else
+          conan remote add --index 0 recipes ${{ env.CONAN_REMOTE_URL }}/recipes
+          echo "Added new conan remote 'recipes' at ${{ env.CONAN_REMOTE_URL }}/recipes."
+        fi
+        echo "Adding Conan remotes for packages."
+        if conan remote list | grep -q "${{ inputs.remote_name }}"; then
+          conan remote update ${{ inputs.remote_name }} --index 1 --url ${{ env.CONAN_REMOTE_URL }}/${{ inputs.remote_name }}
+          echo "Updated Conan remote '${{ inputs.remote_name }}' to ${{ env.CONAN_REMOTE_URL }}/${{ inputs.remote_name }}."
+        else
+          conan remote add --index 1 ${{ inputs.remote_name }} ${{ env.CONAN_REMOTE_URL }}/${{ inputs.remote_name }}
+          echo "Added new conan remote '${{ inputs.remote_name }}' at ${{ env.CONAN_REMOTE_URL }}/${{ inputs.remote_name }}."
         fi
-        conan remote add --index 0 ripple "${CONAN_URL}"
-        echo "Added conan remote ripple at ${CONAN_URL}"
-    - name: try to authenticate to Ripple Conan remote
-      if: env.CONAN_LOGIN_USERNAME_RIPPLE != '' && env.CONAN_PASSWORD_RIPPLE != ''
+    - name: Log into Conan remotes
+      if: env.CONAN_REMOTE_URL != '' && env.CONAN_REMOTE_USERNAME != '' && env.CONAN_REMOTE_PASSWORD != ''
       id: remote
       shell: bash
       run: |
-        echo "Authenticating to ripple remote..."
-        conan remote auth ripple --force
+        conan remote login recipes "${{ env.CONAN_REMOTE_USERNAME }}" --password "${{ env.CONAN_REMOTE_PASSWORD }}"
+        conan remote login ${{ inputs.remote_name }} "${{ env.CONAN_REMOTE_USERNAME }}" --password "${{ env.CONAN_REMOTE_PASSWORD }}"
```

Comment:
assuming we have to login here, would we prefer to use the same style of login as used by `libxrpl` workflow i.e. `conan user --remove .... ` , taking the username and password implicitly from `CONAN_LOGIN_USERNAME` and `CONAN_PASSWORD` ?

Reply:
oh, I did not know that. Thanks !